### PR TITLE
Reader Post Details Comments: add functionality to `Follow Conversation` button

### PR DESCRIPTION
**Pull request by ScoutHarris**
Ref: #17632 

This uses the `ReaderCommentsFollowPresenter` to display the `Follow Conversation` flow from the Comments snippet `Follow Conversation` button.

Notes:
- If following is toggled from the Comments view, the Comments snippet button is not updated on-the-fly yet.

To test:
- Enable the `followConversationPostDetails` feature flag.
- Go to Reader, and select any unsubscribed post.
- 🔍 Verify the "Follow Conversation" button is shown in the Comments header.
- Tap the Follow Conversation button.
- 🔍 Verify that the snackbar with the "Enable" action button is shown.
- Tap on the "Enable" button.
- 🔍 Verify that a feedback notice with an "Undo" action button is shown.
- Tap on the "Undo" action button.
- 🔍 Verify that a feedback notice is shown.
- 🔍 Verify that the Follow Conversation button has turned into a bell icon button.
- Go to Reader, and select any subscribed post.
- Tap on the bell icon button.
- 🔍 Verify that the bottom sheet is shown.
- Tap on the switch.
- Dismiss the bottom sheet.
- 🔍 Verify that a feedback notification is shown, and correctly represents the desired state.
- Tap the bell icon button again.
- Tap on the Unfollow Conversation button.
- 🔍 Verify that a feedback notification is shown.
- 🔍 Verify that the bell icon button has now turned back into "Follow Conversation".

## Regression Notes
1. Potential unintended areas of impact
N/A. Feature is incomplete and disabled.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A. Feature is incomplete and disabled.

3. What automated tests I added (or what prevented me from doing so)
N/A. Feature is incomplete and disabled.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Reader/Comments/ReaderCommentsFollowPresenter.swift [hunk 1]
```diff
@@ -132,9 +132,11 @@ class ReaderCommentsFollowPresenter: NSObject {
...
 
     @objc func showNotificationSheet(sourceBarButtonItem: UIBarButtonItem?) {
-        let sheetViewController = ReaderCommentsNotificationSheetViewController(isNotificationEnabled: post.receivesCommentNotifications, delegate: self)
-        let bottomSheet = BottomSheetViewController(childViewController: sheetViewController)
-        bottomSheet.show(from: self.presentingViewController, sourceBarButtonItem: sourceBarButtonItem)
+        showNotificationSheet(sourceBarButtonItem: sourceBarButtonItem)
+    }
+
+    func showNotificationSheet(sourceView: UIView?) {
+        showNotificationSheet(sourceView: sourceView)
```
Comment (dvdchr) [inline WordPress/Classes/ViewRelated/Reader/Comments/ReaderCommentsFollowPresenter.swift]: This recursively calls the method and caused the app to go into an infinite loop. Strangely enough, `@objc showNotificationSheet(sourceBarButtonItem:)` works just fine – perhaps due to the `@objc` attribute? 🤔 

Some options:
1. Remove the default parameter values from the private methods, so the public method explicitly calls the private method with two parameters.
2. Slightly change the private method name.
3. Make the private method public, but we'd also need to add `@objc` to it.

**Review by dvdchr — CHANGES_REQUESTED** [review]
The app hanged when I tried to tap on the bell icon 😅. tested using iPhone 8 Simulator, iOS 15.2.

File: WordPress/Classes/ViewRelated/Reader/Comments/ReaderCommentsFollowPresenter.swift (hunk 1 above)
Comment (ScoutHarris) [reply WordPress/Classes/ViewRelated/Reader/Comments/ReaderCommentsFollowPresenter.swift]: Um... yea... the private method name was different. Until I decided to make it the same, and then clearly didn't retest. :facepalm: I have renamed it once again. Sorry about that!

**Review by dvdchr — APPROVED** [review]
LGTM! :shipit:

**Comment by ScoutHarris** [comment]
Thanks @dvdchr !

**Pull request merged**